### PR TITLE
base-project-structure

### PR DESCRIPTION
**Pull request by maelstrom0x8**
This is my idea for the projects structure. Here I have defined the data access, presentation and service layer and also made the first step with spring security. I believe we can work from here on to build the future capabilities.

### CODE REVIEW

**Review by minishza — APPROVED** [review]

File: src/main/java/com/jinforatix/app/service/CourseService.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    public List<AuthorDTO> getAllAuthors() {
+        var authors = authorRepository.findAll();
+        return authors.stream()
+            .map(this::convertToDto)
+            .collect(Collectors.toList());
+    }
+
+    public List<CourseDTO> getAllCourses() {
+        var courses = courseRepository.findAll();
+        return courses.stream().map(this::convertToDto).collect(Collectors.toList());
```
Comment (DawidStuzynski) [inline src/main/java/com/jinforatix/app/service/CourseService.java]: You can simplify this with just toList()

Comment (maelstrom0x8) [reply src/main/java/com/jinforatix/app/service/CourseService.java]: Yes. I have done that.

File: src/main/java/com/jinforatix/app/coursemanager/domain/dto/CourseDTO.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    public Instant getLastModified() {
+        return lastModified;
+    }
+
+    public void setLastModified(Instant lastModified) {
+        this.lastModified = lastModified;
+    }
+
+}
```
Comment (DawidStuzynski) [inline src/main/java/com/jinforatix/app/coursemanager/domain/dto/CourseDTO.java:9]: I think this DTO class can be record

Comment (maelstrom0x8) [reply src/main/java/com/jinforatix/app/coursemanager/domain/dto/CourseDTO.java:9]: Done.

**Pull request merged**